plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1357 @@
+        """ The command generator when state is merged
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        test_keys = self.new_test_keys(want)
+        commands = get_diff(want, have, test_keys)
+        request_commands = commands
+        commands_updated = []
+        del_commands = {}
+        delete_user_requests = []
+        requests = []
+        new_have = {}
+        if 'user' in want:
+            delete_user_requests = self.delete_snmp_matched_user_config(commands, have)
+            requests = delete_user_requests
+            del_commands = {'user': self.get_existing_user(commands, have)}
+            new_have = get_diff(have, del_commands)
+            request_commands = get_diff(want, new_have, TEST_KEYS)
+
+        requests.extend(self.get_create_snmp_request(request_commands, new_have, False))
+        if commands and len(requests) > 0:
+            if 'user' in del_commands and len(del_commands['user']) > 0:
+                commands_updated.extend(update_states(del_commands, "deleted"))
+            commands_updated.extend(update_states(request_commands, "merged"))
+        else:
+            commands_updated = []
+
+        return commands_updated, requests
+
+    def _state_deleted(self, want, have):
+        """ The command generator when state is deleted
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        requests = []
+        commands = []
+
+        if not have:
+            return commands, requests
+
+        delete_all = False
+
+        if have is None:
+            return commands, requests
+        if not want or want is None:
+            commands = have
+            delete_all = True
+        else:
+            want = self.get_configured_option(want, have)
+            reverse_diff = get_diff(have, want, TEST_KEYS)
+            commands = get_diff(want, reverse_diff, TEST_KEYS)
+
+        requests = self.get_delete_snmp_request(commands, have, delete_all)
+
+        if len(commands) > 0 and len(requests) > 0:
+            commands = update_states(commands, "deleted")
+        else:
+            commands = []
+
+        return commands, requests
+
+    def get_configured_option(self, want, config):
+        """ Gets the already configured option when want contains an empty list
+
+        :rtype: A dict
+        :returns: the already configured option
+        """
+        new_config = {}
+        want = dict(want)
+        for key, value in want.items():
+            if value == [] and key in config:
+                new_config[key] = config[key]
+            elif 'group' in config and 'group' == key:
+                new_value = []
+                for grp in value:
+                    new_value.append(grp)
+                new_config[key] = new_value
```

Comment:
Why is this "elif" block needed? Wouldn't the same thing happen with the group list by just executing the "else" instruction below for the 'group' key? If that would serve the purpose, please delete this "elif" block.